Series.shift with BDay is not allowed

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**

```python
sr = pd.Series(pd.bdate_range("2023-01-03", "2023-02-06"))
check(assert_type(sr.shift(BDay(1)), "TimestampSeries"), pd.Series, pd.Timestamp)
```

This will raise an issue as only `DateOffset` is allowed in the `period` parameter. The issue is that `BDay` does not inherit from DateOffset but directly from `BaseOffset`.

Is there a reason why we only allow DateOffset and not BaseOffset?

Happy to take care of the PR.

**Please complete the following information:**
 - OS: MacOS
 - OS Version 15.5
 - python version 3.13
 - version of type checker 1.17.0
 - version of installed `pandas-stubs` main

Okay my bad, I was not using the right snippet of code:

```python
import pandas as pd
from pandas.tseries.offsets import BDay

sr = pd.Series([1, 2, 3, 4, 5], index=pd.bdate_range("2023-04-05", periods=5))
sr.shift(freq=BDay(1))
```
will raise the following error (this is the `mypy` error but `pyright` raises the same):
```
tests/test_series.py:4001: error: Argument "freq" to "shift" of "Series" has incompatible type "BusinessDay"; expected "DateOffset | timedelta | str | None"  [arg-type]
Found 1 error in 1 file (checked 208 source files)
```

For the `freq` argument should we allow for `BDay` (which is not a `DateOffset` but inherits from `BaseOffset` if I am correct.

--- a/pandas-stubs/core/frame.pyi
+++ b/pandas-stubs/core/frame.pyi
@@ -892,7 +892,7 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
     def shift(
         self,
         periods: int | Sequence[int] = ...,
-        freq: DateOffset | dt.timedelta | _str | None = ...,
+        freq: BaseOffset | dt.timedelta | _str | None = ...,
         axis: Axis = ...,
         fill_value: Scalar | NAType | None = ...,
     ) -> Self: ...

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -1136,7 +1136,7 @@ class Series(IndexOpsMixin[S1], NDFrame):
     def shift(
         self,
         periods: int | Sequence[int] = ...,
-        freq: DateOffset | timedelta | _str | None = ...,
+        freq: BaseOffset | timedelta | _str | None = ...,
         axis: Axis = ...,
         fill_value: Scalar | NAType | None = ...,
     ) -> Series: ...

--- a/tests/series/test_series.py
+++ b/tests/series/test_series.py
@@ -63,6 +63,10 @@ from tests.extension.decimal.array import DecimalDtype
 from pandas.io.formats.format import EngFormatter
 from pandas.tseries.offsets import (
     BaseOffset,
+    BDay,
+    BQuarterEnd,
+    MonthEnd,
+    Week,
     YearEnd,
 )
 
@@ -448,6 +452,7 @@ def test_types_sort_values_with_key() -> None:
 
 
 def test_types_shift() -> None:
+    """Test shift operator on series with different arguments."""
     s = pd.Series([1, 2, 3], index=pd.date_range("2020", periods=3))
     check(assert_type(s.shift(), pd.Series), pd.Series, np.floating)
     check(
@@ -457,6 +462,11 @@ def test_types_shift() -> None:
     )
     check(assert_type(s.shift(-1, fill_value=0), pd.Series), pd.Series, np.integer)
     check(assert_type(s.shift(freq="1D"), pd.Series), pd.Series, np.integer)
+    check(assert_type(s.shift(freq=BDay(1)), pd.Series), pd.Series, np.integer)
+    check(assert_type(s.shift(freq=BQuarterEnd(5)), pd.Series), pd.Series, np.integer)
+    check(assert_type(s.shift(freq=MonthEnd(3)), pd.Series), pd.Series, np.integer)
+    check(assert_type(s.shift(freq=Week(4)), pd.Series), pd.Series, np.integer)
+    check(assert_type(s.shift(freq=YearEnd(2)), pd.Series), pd.Series, np.integer)
 
 
 def test_series_pct_change() -> None:

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -60,6 +60,13 @@ from tests import (
 from pandas.io.formats.format import EngFormatter
 from pandas.io.formats.style import Styler
 from pandas.io.parsers import TextFileReader
+from pandas.tseries.offsets import (
+    BDay,
+    BQuarterEnd,
+    MonthEnd,
+    Week,
+    YearEnd,
+)
 
 if TYPE_CHECKING:
     from pandas.core.frame import _PandasNamedTuple
@@ -603,6 +610,11 @@ def test_types_shift() -> None:
     check(assert_type(df.shift(1), pd.DataFrame), pd.DataFrame)
     check(assert_type(df.shift(-1), pd.DataFrame), pd.DataFrame)
     check(assert_type(df.shift(freq="1D"), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.shift(freq=BDay(1)), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.shift(freq=BQuarterEnd(5)), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.shift(freq=MonthEnd(3)), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.shift(freq=Week(4)), pd.DataFrame), pd.DataFrame)
+    check(assert_type(df.shift(freq=YearEnd(2)), pd.DataFrame), pd.DataFrame)
 
 
 def test_types_rank() -> None:

